### PR TITLE
Automatically ignore instances based on a regex string

### PR DESCRIPTION
**Pull request by emmasax1**
Description and Impact
----------------------
Automatically ignore certain running instances if they have the words `kitchen` or `auto` in their title. This should print them in blue (like tagged instances), but will say `IGNORED` instead.

Deploy Plan
-----------
- checkout master
- `op accept-pull 26`
- `soyuz deploy`

Rollback Plan
-------------
* To roll back this change, revert the merge with: `git revert -m 1 MERGE_SHA` and perform another deploy.

URLs
----
> Links to bug tickets or user stories.

QA Plan
-------
- [ ] Run `GLI_DEBUG=true bin/sport-ngin-aws-auditor audit [account]` to see ignored instances print in the terminal
- [ ] Run `GLI_DEBUG=true bin/sport-ngin-aws-auditor audit -s [account]` to see ignored instances print into Slack
- [ ] Compare with a current version of master and verify that the ignored instances in the new version are appearing as Missing RIs in the current version

### CODE REVIEW

File: lib/sport_ngin_aws_auditor/instance_helper.rb [hunk 1]
```diff
@@ -144,6 +163,21 @@ def filter_ris_region_based(ris)
       ris.select { |ri| ri.scope == 'Region' }
     end
 
+    # this breaks up the instances array into instances with 'auto' or 'kitchen' in the name and instances without
+    def filter_ignored_instances(instances)
+      instances.partition { |instance| instance.name.include?("auto") || instance.name.include?("kitchen") }
```
Comment (bbergstrom) [inline lib/sport_ngin_aws_auditor/instance_helper.rb]: The list of ignored name patterns should be configurable, not hard coded into the open source gem.

File: lib/sport_ngin_aws_auditor/commands/audit.rb [hunk 2]
```diff
@@ -11,9 +11,9 @@
   c.switch [:n, :no_tag], :desc => "Ignore all tags during audit"
   c.switch [:s, :slack], :desc => "Will print condensed version of audit to a Slack channel"
   c.switch [:z, :zone_output], :desc => "Will print the Missing RIs and Tagged instances with zones"
-  c.flag [:g, :ignore_instances_regex], :default_value => "kitchen, auto", :desc => "Ignore instances if an instance contains
+  c.flag [:g, :ignore_instances_strings], :default_value => "kitchen, auto", :desc => "Ignore instances if an instance contains
```
Comment (bbergstrom) [inline lib/sport_ngin_aws_auditor/commands/audit.rb]: The regex string is called a Pattern.  Lets change to to something like `ignored_instance_patterns` or `instance_ignore_pattern` to be more clear about what this argument is and does.

**Comment by emmasax1** [comment]
### QA:

screen shot 2016-12-12 at 9 28 18 am

**Comment by emmasax1** [comment]
QA approval

**Pull request merged**